### PR TITLE
boot: Fix build time error in DIRECT_XIP mode

### PR DESCRIPTION
**Pull request by Sherryzhang2**
Signed-off-by: Sherry Zhang <sherry.zhang2@arm.com>
Change-Id: I18ce06e008f858a1fef246bf1a4e527320ba6912

### CODE REVIEW

**Comment by davidvincze** [comment]
Could you explain what the build error was? Thank you.

**Comment by Sherryzhang2** [comment]
https://github.com/mcu-tools/mcuboot/issues/1052
The error code is:
`mcuboot/boot/bootutil/src/loader.c:2890:9: error: 'slot_usage' undeclared (first use in this function); did you mean 'slot_usage_t'?
     if (slot_usage[BOOT_CURR_IMG(state)].swap_state.image_ok == BOOT_FLAG_SET) {`

File: boot/bootutil/src/loader.c [hunk 1]
```diff
@@ -2868,12 +2868,14 @@ boot_load_and_validate_images(struct boot_loader_state *state,
...
  * @param  active_slot  Index of the slot will be loaded for current image.
+ * @param  slot_usage   Information about the active and available slots.
  *
  * @return              0 on success; nonzero on failure.
  */
 static int
 boot_update_hw_rollback_protection(struct boot_loader_state *state,
-                                   uint32_t active_slot)
+                                   uint32_t active_slot,
+                                   const struct slot_usage_t slot_usage[])
```
Comment (davidvincze) [inline boot/bootutil/src/loader.c:2878]: Please remove the `active_slot` from the parameter list and extract the active slot index from the the `slot_usage` array inside function.

**Comment by davidvincze** [comment]
> #1052
> The error code is:
> `mcuboot/boot/bootutil/src/loader.c:2890:9: error: 'slot_usage' undeclared (first use in this function); did you mean 'slot_usage_t'? if (slot_usage[BOOT_CURR_IMG(state)].swap_state.image_ok == BOOT_FLAG_SET) {`

Thank you, now I can see that.

File: boot/bootutil/src/loader.c (hunk 1 above)
Comment (Sherryzhang2) [reply boot/bootutil/src/loader.c]: Done

File: boot/bootutil/src/loader.c [hunk 2]
```diff
@@ -2905,7 +2906,7 @@ boot_update_hw_rollback_protection(struct boot_loader_state *state,
 #else /* MCUBOOT_HW_ROLLBACK_PROT */
     (void) (state);
     (void) (active_slot);
```
Comment (davidvincze) [inline boot/bootutil/src/loader.c:2908]: This can be removed.
```suggestion
```

**Review by tamasban — APPROVED** [review]

**Review by davidvincze — APPROVED** [review]

**Pull request merged**